tools/pipelines/test-perf-benchmarks.yml

```diff
@@ -328,6 +334,123 @@ stages:
+        parameters:
+          # ado-feeds-dev and ado-feeds-office come from the ado-feeds variable group
+          artifactBuildId: $(artifactBuildId)
+          artifactPipeline: $(artifactPipeline)
+          devFeedUrl: $(ado-feeds-dev)
+          officeFeedUrl: $(ado-feeds-office)
+          testFilesPath: $(testFilesPath)
+          testWorkspace: $(testWorkspace)
+
+      # Run tests for each package that has them
+      - ${{ each testPackage in parameters.customBenchmarkTestPackages }}:
+
+        # Download and install package with performance tests
+        - template: templates/include-test-perf-benchmarks-install-package.yml
+          parameters:
+            artifactBuildId: $(artifactBuildId)
+            artifactPipeline: $(artifactPipeline)
+            testPackageName: ${{ testPackage }}
+            installPath: $(testWorkspace)
+
+        # The "npm pack"-ed package that we install doesn't have the test files.
+        # Unpack them "on top" of the package, where it was installed, so we can run its tests.
+        - task: Bash@3
+          displayName: Unpack test files - ${{ testPackage }}
+          inputs:
+            targetType: 'inline'
+            script: |
+              TAR_FILENAME=${{ replace(replace(replace(replace(testPackage, '@fluidframework/', '' ), '@fluid-internal/', '' ),'@fluid-', '' ), '/', '-') }}
+              TAR_PATH=$(testFilesPath)/$TAR_FILENAME.test-files.tar
+              cd ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}
+              echo "Unpacking test files for ${{ testPackage }} from file '$TAR_PATH' in '$(pwd)'"
+              # Note: we could skip the last argument and have it unpack everything at once, but if we later change
+              # the structure/contents of the tests tar file, the extraction could overwrite things we didn't intend to,
+              # so keeping the paths to extract explicit.
+              # Also, extracting is finicky with the exact format of the last argument, it needs to match how the
+              # tarfile was created (e.g. './lib/test' works here but 'lib/test' does not).
+              tar --extract --verbose --file $TAR_PATH ./lib/test
+              tar --extract --verbose --file $TAR_PATH ./dist/test
+              tar --extract --verbose --file $TAR_PATH ./src/test
+
+        # Install package dependencies (this gets devDependencies installed so we can then run the package's tests)
+        # and run tests.
+        - task: Bash@3
+          displayName: Run custom data performance tests - ${{ testPackage }}
+          inputs:
+            targetType: 'inline'
+            workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ testPackage }}
+            script: |
+              echo "Run customBenchmark performance test for ${{ testPackage }}"
+              cp ${{ variables.testWorkspace }}/.npmrc . ;
+              npm i ;
+              npm run test:customBenchmarks;
```

Comment:
I refactored how this is done for the existing benchmarks above. Have a look at lines 266-287 and let's do the same thing here. Particularly, we want to have the installation of dependencies be its own step so it can be retried on its own, since it can fail due to transient network errors.

Reply:
Refactored. Thanks!